qiskit/circuit/__init__.py

```diff
@@ -10,38 +10,62 @@
 # copyright notice, and modified files need to carry a notice indicating
 # that they have been altered from the originals.
 
-"""
-========================================
-Quantum Circuits (:mod:`qiskit.circuit`)
-========================================
+r"""
+=============================================
+Quantum circuit model (:mod:`qiskit.circuit`)
+=============================================
 
 .. currentmodule:: qiskit.circuit
 
-Overview
-========
+The fundamental element of quantum computing is the *quantum circuit*.  This is a computational
+routine that can be run, one shot at a time, on a quantum processing unit (QPU).  A circuit will act
+on a predefined amount of quantum data (in Qiskit, we only directly support qubits) with unitary
+operations (gates), measurements and resets.  In addition, a quantum circuit can contain operations
+on classical data, including near-time computations and control-flow constructs, which are executed
+by the controllers of the QPU.
+
+.. note::
+
+    You may wish to skip the introductory material and jump directly to:
+
+    * :ref:`the API overview of the whole circuit module <circuit-module-api>`
+    * :ref:`the detailed discussion about how circuits are represented <circuit-repr>`
+    * the core :class:`QuantumCircuit` class for how to build and query circuits
+    * :ref:`information on construction custom instructions <circuit-custom-gates>`
+    * :ref:`ways to work with circuit-level objects <circuit-working-with>`
+    * :ref:`discussion of Qiskit conventions for circuits, matrices and state labelling
+      <circuit-conventions>`
+
+Circuits are the low-level construct that are used to build up the higher level :ref:`primitives of
+quantum computation <qiskit-primitives>`, which accumulate many shots of quantum-circuit execution
+along with advanced error-mitigation techniques and measurement optimizations.
```

Comment:
```suggestion
In the Qiskit software project, quantum circuits are low-level constructs that are used to build up objects that are on a higher level of abstraction such as :ref:`primitives of
quantum computation <qiskit-primitives>`. Primitives accumulate encapsulate quantum circuit executions with advanced error-mitigation techniques and measurement optimizations.
```

Reply:
I'm not sure your text is quite correct about the minimum requirements of a primitive. The absolute requirement of a primitive is some sort of defined accumulation of several shots into a collection of classical data - all the error mitigation stuff is extras that can be applied but needn't (for example the reference ones don't do any).

Which bit were you aiming to clarify here?

Reply:
I think I wanted to clarify:
- The meaning of 'level': it could mean closer to the hardware, more/less abstract, hierarchical or even good/bad.
- If you don't already know about error-mitigation and measurement optimization you might wonder what "accumulate shots" has to do with the former.
I don't think addressing these is absolutely necessary but I think it would make this bit more informative.

Reply:
Also, I thought it is important to clarify that now we are speaking about circuits as a software object instead of a concept/model of quantum computation...

Reply:
> clarify that now we are speaking about circuits as a software object

Actually, I'd argue that at this point in the text, we're not entirely making that distinction - we do that in the two-to-three paragraphs after this.  I'd argue that this paragraph is still consistent with (somewhat fuzzy) definition of a "circuit" from outside the context of software.

> If you don't already know about error-mitigation and measurement optimization you might wonder what "accumulate shots" has to do with the former.

I'll tweak the text a little to make this clearer, thanks.  I don't think we necessarily need to define all the error-mitigation terms or anything, because they're not directly relevant to the discussion of `qiskit.circuit`; they'd be more appropriate for external learning material and/or the Primitives API documentation, if that needs to refer to them.

Reply:
I've tweaked the terminology around the primitives bit to make it clear that "levels" referred to abstractions, and to expand the detail around how the primitives accumulate bitwise data into well-typed classical data with error statistics in 2c1cc17232a - I think that hopefully addresses your concerns?